test/lib/ansible_test/_data/sanity/validate-modules/validate_modules/main.py

```diff
@@ -486,7 +486,7 @@ def _find_module_utils(self, main):
                     pass
                 names.extend([n.name for n in child.names])
 
-                if [n for n in names if n.startswith('ansible.module_utils')]:
+                if [n for n in names if n.startswith('ansible.module_utils') or (self.collection and 'plugins.module_utils' in n)]:
```

Comment:
It may be worthwhile to split and compare these.  There's nothing stopping a collection from having ```NS.COLLECTN.plugins.modules.plugins.module_utils.MODULENAME``` for instance.  If you split, then you can check that `n.split('.')[2:4] == ['plugins', 'module_utils']`

Reply:
This probably needs to accept any `module_utils` import that Ansible recognizes as a new-style Python module:

https://github.com/ansible/ansible/blob/e4eea0510cd89124c3a20f4d9937e7c015a46597/lib/ansible/executor/module_common.py#L423-L440

Reply:
There is an issue for allowing relative imports at #61884, so-far, this PR does not claim to handle that.

Reply:
That issue is for relative import support in the `import` sanity test.

If we're fixing the `module_utils` matching in `validate-modules` then it should take into account all of the necessary patterns.

Reply:
@abadger Okay, updated the fix to be more defensive.

@mattclay I feel like that is outside of scope of this PR which aims to fix https://github.com/ansible/ansible/issues/62496. However I changed it to account for `'..module_utils'`.